python/pypto/language/parser/ast_parser.py

```
@@ -1060,22 +1060,31 @@ def parse_with_statement(self, stmt: ast.With) -> None:
+
         if isinstance(context_expr, ast.Call):
             func = context_expr.func
-            if isinstance(func, ast.Attribute) and func.attr == "incore":
-                # This is pl.incore()
+            if isinstance(func, ast.Attribute) and func.attr in _SCOPE_KIND_MAP:
+                if context_expr.args or context_expr.keywords:
+                    raise ParserSyntaxError(
+                        f"pl.{func.attr}() does not accept arguments",
+                        span=self.span_tracker.get_span(stmt),
+                        hint=f"Use 'with pl.{func.attr}():' without arguments",
+                    )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)